src/sidecar/connections.ts

```diff
@@ -265,6 +266,9 @@ export async function waitForConnectionToBeUsable(
           continue;
         }
         connectionUsable.fire(id);
+        // notify subscribers that the "environment" has changed since direct connections are treated
+        // as environment-specific resources
+        environmentChanged.fire(id);
```

Comment:
Some overlap with https://github.com/confluentinc/vscode/pull/814 as well since we'll want to fire it in the event of a timeout.